llama_index/chat_engine/types.py

```diff
@@ -1,41 +1,53 @@
 import logging
 import queue
 from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
 from enum import Enum
-from typing import AsyncGenerator, Generator, List, Optional, Union
+from typing import Generator, List, Optional
 
+from llama_index.tools import ToolOutput
 from llama_index.llms.base import ChatMessage, ChatResponseGen, ChatResponseAsyncGen
-from llama_index.response.schema import RESPONSE_TYPE, StreamingResponse
 
 logger = logging.getLogger(__name__)
 
 
-class StreamingChatResponse:
+@dataclass
+class AgentChatResponse:
```

Comment:
does this break backwards compat for existing chat engines/agents in parsing the response type?

Reply:
which backwards compat are you thinking of? The only breaking change I noticed was that agents don't stream a generator of generators anymore, just a single generator for the final response

Reply:
oh mostly that the existing chat engine response type was `RESPONSE_TYPE`, but now it's `AgentChatResponse`. So if the user was doing something like `response.source_nodes` it would break

Reply:
ah yea, that is a possibility 🤔

Reply:
could do a temporary property on AgentChatResponse called `source_nodes` (and try fetching raw_output.source_ndoes..., but seems hacky)

Reply:
I feel like breaking this is inevitable.

Reply:
and I suspect we are probably gonna break it a couple more times as we figure out the interface.